pkg/scd/subscriptions_handler.go

```diff
@@ -305,7 +305,7 @@ func (a *Server) GetSubscription(ctx context.Context, req *restapi.GetSubscripti
 	id, err := dssmodels.IDFromString(string(req.Subscriptionid))
 	if err != nil {
 		return restapi.GetSubscriptionResponseSet{Response400: &restapi.ErrorResponse{
-			Message: dsserr.Handle(ctx, stacktrace.NewErrorWithCode(dsserr.BadRequest, "Invalid ID format: `%s`", req.Subscriptionid))}}
+			Message: dsserr.Handle(ctx, stacktrace.Propagate(err, "Invalid ID format: `%s`", req.Subscriptionid))}}
```

Comment:
Note: this change is not strictly required but I used it to get feedback on the cause of the failure to the caller: we may discuss if we should check and adapt other places that would have hidden the error.

Reply:
Propagating is generally preferred to making a new error, but NewErrorWithCode is where we apply the error type (e.g., BadRequest).  That's not necessarily the best approach, but regarding this change in isolation, are we sure that all `err`s we may propagate here will originate with `dsserr.BadRequest`, or are we sure we will always indicate BadRequest further up the stack?

Reply:
I'll revert this:
 - the main purpose was for troubleshooting and confirmation that the fix works as expected from the point of view of the `uss_qualifier`.
 - It would be hard to guarantee that the error type would be correct in all situations, or it would require more work that does not fall within the current scope and priorities.